selfdrive/car/docs_definitions.py

```diff
@@ -38,6 +38,130 @@ class PartType(Enum):
   mount = "Mount"
 
 
+# A part + its comprised parts
+@dataclass
+class BasePart:
+  name: str
+  # A list of parts that come with this part in the shop
+  parts: List[Enum] = field(default_factory=list)
+  # If there are extra parts included, you can override required_parts
+  required_parts: Optional[List[Enum]] = None
+
+  def __post_init__(self):
+    if self.required_parts is None:
+      self.required_parts = self.parts
+
+  def all_parts(self, required=False):
+    # Recursively get all parts
+    _parts = 'required_parts' if required else 'parts'
+
+    parts = []
+    parts.extend(getattr(self, _parts))
+    for part in getattr(self, _parts):
+      parts.extend(part.value.all_parts(required))
+
+    return parts
+
+
+class EnumBase(Enum):
+  @classmethod
+  def type(cls):
+    return PartTypeNew(cls)
```

Comment:
this allows similar type checking in the Jinja template as before (no isinstance required)